2021/day14/day14.java

```diff
@@ -0,0 +1,89 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.io.File;
+
+public class day14 {
+    public static class Rules {
+        String pair;
+        String single;
+    }
+
+    public static String readTemplate(String filename) {
+        String template = null;
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            template = sc.next();
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return template;
+    }
+
+    public static ArrayList<Rules> readRules(String filename) {
+        Scanner sc = null;
+        ArrayList<Rules> input = new ArrayList<>();
+        try {
+            sc = new Scanner(new File(filename));
+            sc.useDelimiter("\\n| -> ");
+            while (sc.hasNext()) {
+                Rules line = new Rules();
+                line.pair = sc.next();
+                line.single = sc.next();
+                input.add(line);
+            }
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return input;
+    }
+
+    public static HashMap<String, Long> splitToChunks(String template) {
+        HashMap<String, Long> chunks = new HashMap<>();
+        for (int i = 0; i < template.length() - 1; i++) {
+            chunks.merge(String.valueOf(template.charAt(i))
```

Comment:
`template.substring(i, i + 2)` instead of these `valueOf(charAt)`s?

Reply:
I did not know this method exists :+1: